ui/packages/catalog-ui-search/src/main/webapp/react-component/location/bounding-box.js

```diff
@@ -30,22 +28,41 @@ const { Direction } = require('../../component/location-new/utils/dms-utils.js')
 
 const minimumDifference = 0.0001
 
-const BoundingBoxLatLon = props => {
-  const { north, east, south, west, cursor } = props
-
-  const { mapEast, mapWest, mapSouth, mapNorth } = props
-
+const BoundingBoxLatLonDd = props => {
+  const {
+    north,
+    east,
+    south,
+    west,
+    setState,
+    mapEast,
+    mapWest,
+    mapSouth,
+    mapNorth,
+  } = props
+  const [ddError, setDdError] = useState({
+    error: false,
+    message: '',
+    defaultValue: '',
+  })
   const westMax = parseFloat(mapEast) - minimumDifference
   const eastMin = parseFloat(mapWest) + minimumDifference
   const northMin = parseFloat(mapSouth) + minimumDifference
   const southMax = parseFloat(mapNorth) - minimumDifference
-
+  function onChangeDd(key, value) {
+    const { error, message, defaultValue } = validateGeo(key, value)
+    if (defaultValue) { setDdError({ error, message, defaultValue }) }
+    defaultValue
+      ? setState(key, defaultValue)
+      : setState(key, value)
+  }
   return (
     <div className="input-location">
       <TextField
         label="West"
-        value={west}
-        onChange={cursor('west')}
+        value={west !== undefined ? String(west) : west}
+        onChange={value => onChangeDd('west', value)}
+        onBlur={() => setDdError(validateGeo('west', west))}
```

Comment:
I'd say let's pass `'lon'` here to simplify the `validateGeo` function

Reply:
good call